confgenerator/config.go

```diff
@@ -149,6 +153,24 @@ func newValidator() *validator.Validate {
 		}
 		return t >= tmin
 	})
+	// time_denomination validates that the duration is whole to some time denomination
```

Comment:
I'm not sure how this would look when specifying the validation on the struct, since 1 second in nanoseconds is 1e+9. I could probably impose some kind of rule like having it work in seconds or milliseconds instead, but I worry that could become confusing for usage.

Reply:
I meant something like calling the validator `multipleof=1s`; you'd probably have to specialize to calling `time.ParseDuration` if the field is a `Duration`; then you can do the actual math on an arbitrary int value.

Reply:
Oh okay I see, that is sort of what I ended up doing. I made a single purpose version that is only for time values rather than making a validator that checks and parses either time or int. I thought it was easier to read that way, and we don't currently have a need for a `multipleof` validator on just ints. I can change it to be that way instead though if that's better.

Reply:
Nah, I was just trying to code golf by keeping the name short and simple :) It's not very important.

Reply:
Looks like this one has been resolved. Thanks!